torch/csrc/jit/tensorexpr/loopnest.cpp

```diff
@@ -3390,6 +3391,732 @@ bool LoopNest::rfactor(
+  // When seed value is 1, we perform preset loop transformations. This allows
+  // testing of interface.
+  if (seed == 1) {
+    simplify();
+    return;
+  }
+
+  std::default_random_engine random_engine(seed);
+  std::srand(seed);
+  int n_transforms =
+      randomization_helper::max_transformations(std::rand() % 20);
+  std::string message = "";
+  // clang-format off
+  //   Transformations list:
+  //
+  //       StmtPtr simplify();
+  //       bool computeInline(BufPtr b);
+  //       void inlineIntermediateBufs(bool allow_duplicated_work);
+  //       bool optimizeConditionals();
+  //       static void splitWithTail(ForPtr f, int factor);
+  //       static void splitWithMask(ForPtr f, int factor);
+  //       static std::vector<ForPtr> distributeLoop(ForPtr loop, const std::unordered_set<StmtPtr>& pivots);
+  //       static std::vector<ForPtr> distributeLoop(ForPtr loop);
+  //       static std::vector<ForPtr> distributeLoopAndParents(ForPtr loop);
+  //       static std::vector<ForPtr> distributeLoopOverInnerLoops(ForPtr loop);
+  //       static std::vector<ForPtr> distributeLoopAndParentsOverInnerLoops(ForPtr loop);
+  //       static bool fuseLoops(const std::vector<ForPtr>& loops, ForPtr* fused);
+  //       static void reorderAxis(ForPtr a, ForPtr b);
+  //       static std::vector<ForPtr> reorder(const std::vector<ForPtr>& loops, const std::vector<size_t>& permutation);
+  //       ForPtr tile(ForPtr x, ForPtr y, int x_factor, int y_factor);
+  //       static void unroll(ForPtr f);
+  //       static bool normalize(ForPtr f);
+  //       static bool flatten(const std::vector<ForPtr>& f, ForPtr* flattened);
+  //       static void compressBuffer(BufPtr buf, StmtPtr stmt);
+  //       static void compressAllBuffers(StmtPtr stmt);
+  //       static void sliceHead(ForPtr f, int factor, ForPtr* head, ForPtr* tail);
+  //       static void sliceHead(ForPtr f, int factor);
+  //       static void sliceTail(ForPtr f, int factor, ForPtr* head, ForPtr* tail);
+  //       static void sliceTail(ForPtr f, int factor);
+  //       static AccessResult cacheAccesses(BufPtr producer, const std::string& name, StmtPtr consumer);
+  //       static void computeAt(StmtPtr s, ForPtr at);
+  //       static bool rfactor(StmtPtr s, ForPtr outer_reduction_for);
+  //       static bool vectorize(ForPtr);
+  //       void vectorizeInnerLoops();
+  //       void eliminateDeadStores();
+  //       void prepareForCodegen();
+  // clang-format on
+  enum TransformKind {
+    SIMPLIFY = 0,
+    COMPUTE_INLINE,
+    INLINE_ALL,
+    OPT_COND,
+    SPLIT_TAIL,
+    SPLIT_MASK,
+    DIST1,
+    DIST2,
+    DIST3,
+    DIST4,
+    DIST5,
+    FUSE_LOOPS,
+    REORDER_AXIS,
+    REORDER,
+    TILE,
+    UNROLL,
+    NORMALIZE,
+    FLATTEN,
+    COMPRESS_BUFFER,
+    COMPRESS_ALL_BUFFERS,
+    SLICE_HEAD, // 20
+    SLICE_TAIL,
+    CACHE_ACCESSES,
+    COMPUTE_AT, // better message
+    RFACTOR, // TODO
```

Comment:
Do you mind expanding or removing these comments? Hard to tell what they mean

Reply:
Done